payjoin/src/receive/v2/mod.rs

```diff
@@ -1,40 +1,46 @@
 use std::collections::HashMap;
-use std::fmt;
 use std::str::FromStr;
 use std::time::{Duration, SystemTime};
 
-use bitcoin::address::NetworkUnchecked;
 use bitcoin::base64::prelude::BASE64_URL_SAFE_NO_PAD;
 use bitcoin::base64::Engine;
 use bitcoin::psbt::Psbt;
-use bitcoin::secp256k1::{rand, PublicKey};
 use bitcoin::{Address, Amount, FeeRate, OutPoint, Script, TxOut};
-use serde::de::{self, Deserializer, MapAccess, Visitor};
-use serde::ser::SerializeStruct;
-use serde::{Deserialize, Serialize, Serializer};
+use serde::de::Deserializer;
+use serde::{Deserialize, Serialize};
 use url::Url;
 
 use super::v2::error::{InternalSessionError, SessionError};
 use super::{Error, InternalRequestError, RequestError, SelectionError};
 use crate::psbt::PsbtExt;
 use crate::receive::optional_parameters::Params;
-use crate::v2::OhttpEncapsulationError;
+use crate::v2::{HpkePublicKey, HpkeSecretKey, OhttpEncapsulationError};
 use crate::{OhttpKeys, PjUriBuilder, Request};
 
 pub(crate) mod error;
 
 static TWENTY_FOUR_HOURS_DEFAULT_EXPIRY: Duration = Duration::from_secs(60 * 60 * 24);
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 struct SessionContext {
+    #[serde(deserialize_with = "deserialize_address_assume_checked")]
     address: Address,
     directory: url::Url,
     subdirectory: Option<url::Url>,
     ohttp_keys: OhttpKeys,
     expiry: SystemTime,
     ohttp_relay: url::Url,
-    s: bitcoin::secp256k1::Keypair,
-    e: Option<bitcoin::secp256k1::PublicKey>,
+    s: (HpkeSecretKey, HpkePublicKey),
```

Comment:
It'd be nice to have a `struct HpkeKeyPair(HpkeSecretKey, HpkePublicKey)` abstraction with `secret/public_key()` methods for clarity.